homeassistant/components/dublin_bus_transport/sensor.py

```diff
@@ -117,64 +100,35 @@ def icon(self):
         return ICON
 
     def update(self):
-        """Get the latest data from opendata.ch and update the states."""
+        """Get the latest data and update the states."""
         self.data.update()
         self._times = self.data.info
-        try:
-            self._state = self._times[0][ATTR_DUE_IN]
-        except TypeError:
-            pass
+        self._state = self._times[0][ATTR_DUE_IN]
 
 
 class PublicTransportData:
     """The Class for handling the data retrieval."""
 
-    def __init__(self, stop, route):
+    def __init__(self, stopid, route):
         """Initialize the data object."""
-        self.stop = stop
+        self.stopid = stopid
         self.route = route
-        self.info = [{ATTR_DUE_AT: "n/a", ATTR_ROUTE: self.route, ATTR_DUE_IN: "n/a"}]
+        self.info = [{ATTR_DUE_IN: "n/a", ATTR_ROUTE: self.route}]
 
     def update(self):
-        """Get the latest data from opendata.ch."""
-        params = {}
-        params["stopid"] = self.stop
-
-        if self.route:
-            params["routeid"] = self.route
-
-        params["maxresults"] = 2
-        params["format"] = "json"
-
-        response = requests.get(_RESOURCE, params, timeout=10)
-
-        if response.status_code != HTTP_OK:
-            self.info = [
-                {ATTR_DUE_AT: "n/a", ATTR_ROUTE: self.route, ATTR_DUE_IN: "n/a"}
-            ]
-            return
-
-        result = response.json()
-
-        if str(result["errorcode"]) != "0":
-            self.info = [
-                {ATTR_DUE_AT: "n/a", ATTR_ROUTE: self.route, ATTR_DUE_IN: "n/a"}
-            ]
-            return
-
+        """Get the latest data from the _RESOURCE."""
+        mybus = DublinBusRTPI(stopid=self.stopid, route=self.route)
```

Comment:
This should be moved to the setup, so you do not create a new object on every update.